wire-ios/Wire-iOS/Sources/UserInterface/Settings/Backup/BackupLocalStore/BackupLocalStore.swift

```diff
@@ -24,14 +24,49 @@ import WireDomain
 struct BackupLocalStore: BackupLocalStoreProtocol, @unchecked Sendable {
 
     /// The context to call `perform(schedule:_:)` on if needed.
-    let context: NSManagedObjectContext
-    let processor: any ConversationProtobufMessageProcessorProtocol
+    let backupContext: NSManagedObjectContext
+    let contextProvider: ContextProvider
+    let assetTransferStateResolver: AssetTransferStateResolverProtocol
+    var clientMessageEntityDescription: NSEntityDescription?
+    var assetMessageEntityDescription: NSEntityDescription?
+
+    init(
+        contextProvider: ContextProvider,
+        assetTransferStateResolver: AssetTransferStateResolverProtocol = AssetTransferStateResolver()
+    ) {
+        self.contextProvider = contextProvider
+        self.assetTransferStateResolver = assetTransferStateResolver
+        self.backupContext = contextProvider.newBackgroundContext()
+
+        setupBackupContext()
+
+        self.clientMessageEntityDescription = NSEntityDescription.entity(
+            forEntityName: ZMClientMessage.entityName(),
+            in: backupContext
+        )
+        self.assetMessageEntityDescription = NSEntityDescription.entity(
+            forEntityName: ZMAssetClientMessage.entityName(),
+            in: backupContext
+        )
+    }
+
+    private func setupBackupContext() {
+
+        // Ensures imported data overwrites existing values without validation conflicts.
+        backupContext.mergePolicy = NSMergeByPropertyObjectTrumpMergePolicy
+
+        // Prevents automatic UI context merges that could slow down the import.
+        backupContext.automaticallyMergesChangesFromParent = false
```

Comment:
I'm not an expert but the combination looks dangerous to me.

1. We create a new context (which is usually a good idea) while there exist other contexts with potentially unsaved changes and then 
2. we set `automaticallyMergesChangesFromParent` to `false` and `mergePolicy` to `NSMergeByPropertyObjectTrumpMergePolicy`.

Maybe @johnxnguyen and/or @netbe could verify that this is a valid approach?